backend/src/ee/services/dynamic-secret/providers/aws-elasticache.ts

```diff
@@ -0,0 +1,105 @@
+import handlebars from "handlebars";
+import { customAlphabet } from "nanoid";
+
+import { CreateElastiCacheUserSchema, DeleteElasticCacheUserSchema, ElastiCacheUserManager } from "@app/lib/aws";
+import { BadRequestError } from "@app/lib/errors";
+
+import { DynamicSecretAwsElastiCacheSchema, TDynamicProviderFns } from "./models";
+
+const generatePassword = () => {
+  const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_.~!*$#";
+  return customAlphabet(charset, 64)();
+};
+
+const generateUsername = () => {
+  const charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-";
+  return `inf-${customAlphabet(charset, 32)()}`; // Username must start with an ascii letter, so we prepend the username with "inf-"
+};
+
+export const AwsElastiCacheDatabaseProvider = (): TDynamicProviderFns => {
+  const validateProviderInputs = async (inputs: unknown) => {
+    const providerInputs = DynamicSecretAwsElastiCacheSchema.parse(inputs);
+
+    JSON.parse(providerInputs.creationStatement);
```

Comment:
Also can we please keep everything related to dynamic secret here. Keeping it lib is really confusing at the moment. I had to jump between two files to understand the logic and it seems quite hard to understand. 

I think if we keep those functions here it will much lean to understand?

Reply:
> I don't see where we are using this 🤔 

We need to ensure the that the creation & revocation statements are valid and can be used to create and revoke users. We can't return the parsed statements here because we need to use the handlebars template to generate the username and password, before we can use the parsed statements

> Also can we please keep everything related to dynamic secret here. Keeping it lib is really confusing at the moment. I had to jump between two files to understand the logic and it seems quite hard to understand.
I think if we keep those functions here it will much lean to understand?

Done!